runscope/runscope.py

```diff
@@ -0,0 +1,157 @@
+
+def is_cache_stale():
+    if os.path.exists(cache_file):
+       file_mod_time = datetime.fromtimestamp(os.stat(cache_file).st_mtime)
+       now = datetime.today()
+       max_cache_time = timedelta(minutes=cache_expiry_min)
+       if now-file_mod_time < max_cache_time:
+          return False
+    return True
+   
+def get_cached_test_meta_info():
+    try :
+        with open(cache_file) as test_mi:
+            if test_mi is None:
+                return None
+            else:
+                mi = json.load(test_mi)
+                test_mi.close()
+                return mi
+    except ValueError as ve:
+        return None
+    except Exception as e:
+        handle_error("error reading from cache: " + str(e))
+
+def handle_error(error_message):
+    sys.stderr.write("ERROR:|Runscope| " + error_message)
+    sys.exit(1)
+    
+def get_runscope_api_response(url):
+    try :
+        headers = {}
+        headers["Authorization"] = "Bearer {} ".format(args.token[0])
+        headers["Content-type"] = "application/json"
+        headers["Accept"] = "application/json"
+        response = requests.get(url,headers=headers)
+        if response.status_code == 200:
+            return json.loads(response.text)
+        else:
+            em = "error connecting to runscope {}".format(response.status_code)
+            handle_error(em)       
+    except Exception as e:
+        handle_error("error while conneecting to runscope: " + str(e))        
+
+def get_test_bucket_details():
+    test_bucket_details = []
+    response = get_runscope_api_response(runscope_service_url)
+    for data in response['data']:
+        test_bucket_meta_info = {}
+        test_bucket_meta_info['bucket_name'] = data['name']
+        test_bucket_meta_info['tests_url'] = data['tests_url']
+        test_bucket_details.append(test_bucket_meta_info)
+    return test_bucket_details
+
+def get_test_details(test_bucket_list):
+    test_details = []
+    for bucket in test_bucket_list:
+        response = get_runscope_api_response(bucket['tests_url'])
+        for test in response['data']:
+            test_meta_info = {}
+            test_meta_info['latest_result_url'] = bucket['tests_url'] + "/{}/results/latest".format(test['last_run']['test_uuid'])
+            test_meta_info['test_name'] = test['name']
+            test_meta_info['test_bucket_name'] = bucket['bucket_name']
+            test_details.append(test_meta_info)
+    return test_details
+
+def get_test_latest_result(test_details):
+    uber_result = []
+    for test in test_details:
+        response = get_runscope_api_response(test['latest_result_url'])
+        for request in response['data']['requests']:
+            service_metric = {}
+            service_url = request['url']
+            if service_url is not None:
+               #Need to check on the tag length         
+              # request_result = str(request['result'])
+               service_metric['service_url'] = service_url[:240]
```

Comment:
You may hit a `LookupError` or such if any of these values are not present in the test/request dictionaries. Even if it works now, it may change in the future. Maybe write a small helper method that takes `service_metric`, `input_dictionary` and `key` and adds it to the service_metric dictionary if the dictionary has the key and value is not None?